### PR TITLE
Refs #169: Add support for paginating permissions.

### PR DESCRIPTION
**Pull request by n1k0**
Refs #169. Add support for paginating the list of permissions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1275,6 +1275,13 @@ Sample result:
 - `headers`: Custom headers object to send along the HTTP request
 - `retry`: Number of retries when request fails (default: 0)
 
+#### Result object properties
+
+- `last_modified`: the last modified value for the list of permissions
+- `next`: the [pagination](#paginating-results) helper to access the next page of results, if any
+- `data`: the list of permissions
+- `totalRecords`: the total number of permissions listed in the `data` property array
```
Comment (leplatrem) [inline README.md]: I think there is a `hasNextPage` boolean or something similar, no?

File: README.md [hunk 2]
```diff
@@ -1275,6 +1275,13 @@ Sample result:
 - `headers`: Custom headers object to send along the HTTP request
 - `retry`: Number of retries when request fails (default: 0)
 
+#### Result object properties
+
+- `last_modified`: the last modified value for the list of permissions
+- `next`: the [pagination](#paginating-results) helper to access the next page of results, if any
```
Comment (leplatrem) [inline README.md]: helper → function (not sure?)

File: src/base.js [hunk 3]
```diff
@@ -455,8 +455,12 @@ export default class KintoClientBase {
    */
   @capable(["permissions_endpoint"])
   listPermissions(options = {}) {
+    const path = endpoint("permissions");
     const reqOptions = this._getRequestOptions(options);
-    return this.execute({ path: endpoint("permissions"), ...reqOptions });
+    // Ensure the default sort parameter is something that exists in permissions
+    // entries, as `last_modified` doesn't; here, we pick "id".
+    const paginationOptions = { sort: "id", ...options };
```
Comment (leplatrem) [inline src/base.js]: You need that for tests reproductibility?

File: test/integration_test.js [hunk 4]
```diff
@@ -245,17 +245,36 @@ describe("Integration tests", function() {
...
+        beforeEach(() => {
+          return api.batch(batch => {
+            for (let i = 1; i <= 15; i++) {
+              batch.createBucket("b" + i);
+            }
+          });
+        });
+
+        it("should retrieve the list of permissions", () => {
+          return api.listPermissions({ pages: Infinity }).then(results => {
```
Comment (leplatrem) [inline test/integration_test.js]: server pagination is less than 15?

File: README.md (hunk 2 above)
Comment (n1k0) [inline README.md]: I just reused what we used for listRecords. helper sounds ok to me.

File: README.md (hunk 1 above)
Comment (n1k0) [inline README.md]: True!

File: src/base.js (hunk 3 above)
Comment (n1k0) [inline src/base.js]: No, to avoid falling into https://github.com/Kinto/kinto/issues/1157

File: test/integration_test.js (hunk 4 above)
Comment (n1k0) [inline test/integration_test.js]: Yes, test server is configured with `paginate_by` set to `10`.

**Pull request merged**